### ISSUE TITLE
Make ROLLUP_USAGE a Configuration Option

### ISSUE THREAD

**Issue by overstre** [OpenTSDB/opentsdb#1080]
It would be useful if ROLLUP_USAGE could be made a configuration option.  The default in TsdbQuery.java is
```
/** How to use the rollup data */
  private ROLLUP_USAGE rollup_usage = ROLLUP_USAGE.ROLLUP_NOFALLBACK;
```
There are many cases when having a different default would make sense such as a sparse rollup table where the default might be rollup_fallback_raw.

**Comment by tkrajca** [comment]
+1

I am also wondering why ROLLUP_NOFALLBACK is the default. I think that makes it hard going from no rollups to rollups as all rollup tables have to be populated prior to switching over. Otherwise, all queries that work before the switchover return no data after the switchover. It would make much more sense to me to have ROLLUP_RAW as the default so that all reads against 2.3 (without rollups) return same results for 2.4 (with rollups enabled). It also makes it trickier for metrics that don't need to use rollups as rollup_raw has to be explicitly added to all such queries.

Finally, I am not quite sure what the difference is between ROLLUP_FALLBACK and ROLLUP_FALLBACK_RAW, they seem to behave identically.

Thanks,

**Comment by manolama** [comment]
Makes sense. We just had it default to the no-fallback as it can be incredibly expensive if you don't have a query limit set.
